hs_app_timeseries/models.py

```diff
@@ -228,96 +498,348 @@ def get_xml(self, pretty_print=True):
-            hsterms_status.text = self.time_series_result.status
-
-            hsterms_sample_medium = etree.SubElement(hsterms_time_series_result_rdf_Description, '{%s}SampleMedium' % self.NAMESPACES['hsterms'])
-            hsterms_sample_medium.text = self.time_series_result.sample_medium
-
-            hsterms_value_count = etree.SubElement(hsterms_time_series_result_rdf_Description, '{%s}ValueCount' % self.NAMESPACES['hsterms'])
-            hsterms_value_count.text = str(self.time_series_result.value_count)
-
-            hsterms_statistics = etree.SubElement(hsterms_time_series_result_rdf_Description, '{%s}AggregationStatistic' % self.NAMESPACES['hsterms'])
-            hsterms_statistics.text = self.time_series_result.aggregation_statistics
+            self.add_metadata_element_to_xml(hsterms_time_series_result_rdf_Description,
+                                             (processing_level, 'processingLevel'),
+                                             element_fields)
 
         return etree.tostring(RDF_ROOT, pretty_print=True)
 
     def delete_all_elements(self):
         super(TimeSeriesMetaData, self).delete_all_elements()
-        if self.site:
-            self.site.delete()
-        if self.variable:
-            self.variable.delete()
-        if self.method:
-            self.method.delete()
-        if self.processing_level:
-            self.processing_level.delete()
-        if self.time_series_result:
-            self.time_series_result.delete()
-
-
-import receivers
+        # delete resource specific metadata
+        self.sites.delete()
+        self.variables.delete()
+        self.methods.delete()
+        self.processing_levels.delete()
+        self.time_series_results.delete()
+        # delete CV lookup django tables
+        self.cv_variable_types.all().delete()
+        self.cv_variable_names.all().delete()
+        self.cv_speciations.all().delete()
+        self.cv_elevation_datums.all().delete()
+        self.cv_site_types.all().delete()
+        self.cv_method_types.all().delete()
+        self.cv_units_types.all().delete()
+        self.cv_statuses.all().delete()
+        self.cv_mediums.all().delete()
+        self.cv_aggregation_statistics.all().delete()
+
+    def update_sqlite_file(self):
+        if not self.is_dirty:
+            return
+
+        sqlite_file_to_update = self.resource.files.first()
+        if sqlite_file_to_update is not None:
+            istorage = IrodsStorage()
+            log = logging.getLogger()
+
+            # retrieve the sqlite file from iRODS and save it to temp directory
+            temp_dir = tempfile.mkdtemp()
+            sqlite_file_name = os.path.basename(sqlite_file_to_update.resource_file.name)
+            temp_sqlite_file_destination = os.path.join(temp_dir, sqlite_file_name)
+            istorage.getFile(sqlite_file_to_update.resource_file.name, temp_sqlite_file_destination)
```

Comment:
@pkdash This does not take into account iRODS federation. What if the sqlite file is stored in a federated zone? You can refer to https://github.com/hydroshare/hydroshare/blob/develop/hs_app_timeseries/receivers.py#L15...L29 to see how that method is used to retrieve a resource file from iRODS taking irods federation into account. You may be able to call that function directly.